diff --git a/src/sched/sched_getparam.c b/src/sched/sched_getparam.c
index 76f10e49..65be1075 100644
--- a/src/sched/sched_getparam.c
+++ b/src/sched/sched_getparam.c
@@ -1,8 +1,7 @@
 #include <sched.h>
-#include <errno.h>
 #include "syscall.h"
 
 int sched_getparam(pid_t pid, struct sched_param *param)
 {
-	return __syscall_ret(-ENOSYS);
+	return syscall(SYS_sched_getparam, pid, param);
 }
diff --git a/src/sched/sched_getscheduler.c b/src/sched/sched_getscheduler.c
index 394e508b..4c922f69 100644
--- a/src/sched/sched_getscheduler.c
+++ b/src/sched/sched_getscheduler.c
@@ -1,8 +1,7 @@
 #include <sched.h>
-#include <errno.h>
 #include "syscall.h"
 
 int sched_getscheduler(pid_t pid)
 {
-	return __syscall_ret(-ENOSYS);
+	return syscall(SYS_sched_getscheduler, pid);
 }
diff --git a/src/sched/sched_setparam.c b/src/sched/sched_setparam.c
index 18623ee4..07d61aea 100644
--- a/src/sched/sched_setparam.c
+++ b/src/sched/sched_setparam.c
@@ -1,8 +1,8 @@
 #include <sched.h>
-#include <errno.h>
 #include "syscall.h"
 
 int sched_setparam(pid_t pid, const struct sched_param *param)
 {
-	return __syscall_ret(-ENOSYS);
+	static const struct sched_param def;
+	return syscall(SYS_sched_setparam, pid, &def);
 }
diff --git a/src/sched/sched_setscheduler.c b/src/sched/sched_setscheduler.c
index 4435f216..19580660 100644
--- a/src/sched/sched_setscheduler.c
+++ b/src/sched/sched_setscheduler.c
@@ -1,8 +1,8 @@
 #include <sched.h>
-#include <errno.h>
 #include "syscall.h"
 
 int sched_setscheduler(pid_t pid, int sched, const struct sched_param *param)
 {
-	return __syscall_ret(-ENOSYS);
+	static const struct sched_param def;
+	return syscall(SYS_sched_setscheduler, pid, 0, &def);
 }
