diff --git zfs-build/include/sys/multilist.h.orig zfs-build/include/sys/multilist.h
index 4395406..d36866e 100644
--- zfs-build/include/sys/multilist.h.orig
+++ zfs-build/include/sys/multilist.h
@@ -47,7 +47,7 @@ struct multilist_sublist {
 	 * Pad to cache line, in an effort to try and prevent cache line
 	 * contention.
 	 */
-} ____cacheline_aligned;
+};
 
 struct multilist {
 	/*
diff --git zfs-build/lib/libshare/smb.c.orig zfs-build/lib/libshare/smb.c
index 76145bd..82ee37a 100644
--- zfs-build/lib/libshare/smb.c.orig
+++ zfs-build/lib/libshare/smb.c
@@ -63,6 +63,7 @@
 static boolean_t smb_available(void);
 
 static sa_fstype_t *smb_fstype;
+smb_share_t *smb_shares;
 
 /*
  * Retrieve the list of SMB shares.
diff --git zfs-build/lib/libshare/smb.h.orig zfs-build/lib/libshare/smb.h
index 7a0c0fd..8ea4467 100644
--- zfs-build/lib/libshare/smb.h.orig
+++ zfs-build/lib/libshare/smb.h
@@ -44,6 +44,6 @@ typedef struct smb_share_s {
 	struct smb_share_s *next;
 } smb_share_t;
 
-smb_share_t *smb_shares;
+extern smb_share_t *smb_shares;
 
 void libshare_smb_init(void);
diff --git zfs-build/cmd/zfs/zfs_util.h.orig zfs-build/cmd/zfs/zfs_util.h
index 3ddff9e..a56af59 100644
--- zfs-build/cmd/zfs/zfs_util.h.orig
+++ zfs-build/cmd/zfs/zfs_util.h
@@ -33,7 +33,7 @@ extern "C" {
 
 void * safe_malloc(size_t size);
 void nomem(void);
-libzfs_handle_t *g_zfs;
+extern libzfs_handle_t *g_zfs;
 
 #ifdef	__cplusplus
 }
diff --git zfs-build/cmd/zpool/zpool_util.h.orig zfs-build/cmd/zpool/zpool_util.h
index ab7662c..1dd93fe 100644
--- zfs-build/cmd/zpool/zpool_util.h.orig
+++ zfs-build/cmd/zpool/zpool_util.h
@@ -73,7 +73,7 @@ void pool_list_free(zpool_list_t *);
 int pool_list_count(zpool_list_t *);
 void pool_list_remove(zpool_list_t *, zpool_handle_t *);
 
-libzfs_handle_t *g_zfs;
+extern libzfs_handle_t *g_zfs;
 
 
 typedef	struct vdev_cmd_data
diff --git zfs-build/cmd/zpool/zpool_main.c.orig zfs-build/cmd/zpool/zpool_main.c
index edcfe2d..2f194bf 100644
--- zfs-build/cmd/zpool/zpool_main.c.orig
+++ zfs-build/cmd/zpool/zpool_main.c
@@ -64,6 +64,8 @@
 
 #include "statcommon.h"
 
+libzfs_handle_t *g_zfs;
+
 static int zpool_do_create(int, char **);
 static int zpool_do_destroy(int, char **);
 
